var/spack/repos/builtin/packages/graphviz/package.py

```diff
@@ -34,11 +34,13 @@ class Graphviz(AutotoolsPackage):
 
     version('2.38.0', '5b6a829b2ac94efcd5fa3c223ed6d3ae')
 
+    # Swig can be enabled on it's own, though it doesn't do much.  It
+    # will also be enabled if any of the languages are enabled.
+    variant('swig', default=False,
+            description='Enable swig language binding tools')
```

Comment:
That makes sense to me.

It's worse than you describe.  `graphviz~swig+python` will pass `--enable-swig` to *configure* but will not include the dependency at the Spack level.  On the one hand you'll end up using your system's swig, on the other you'll crash for lack of a swig (and on the gripping hand you'll trip over your system's broken swig in `/usr/local`, sigh...).

I was trying to minimize my impositions.